ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/offset/KudafByOffsetUtils.java

```diff
@@ -64,6 +64,21 @@
     }
   };
 
+  static final Comparator<Struct> INTERMEDIATE_STRUCT_COMPARATOR_IGNORE_NULLS =
+      (struct1, struct2) -> {
+        // Ignore nulls: If one of the structs has a null value, then return the other irrespective
+        // of sequence.
+        if (struct1.get(VAL_FIELD) == null) {
+          return -1;
+        } else if (struct2.get(VAL_FIELD) == null) {
+          return 1;
+        } else if (struct1.get(VAL_FIELD) == null && struct2.get(VAL_FIELD) == null) {
+          return 0;
+        }
```

Comment:
this condition will never be hit - it needs to be the first in the if/else branches

Reply:
These are what happens when you work on the 3 PRs in parallel and the release